### ISSUE TITLE
FluxImg2ImgPipeline produces output almost identical to the input image

### ISSUE THREAD

**Issue by CyberVy** [huggingface/diffusers#10210]
### Describe the bug

When providing an input image and a prompt, the resulting output image is nearly identical to the input image, regardless of how I adjust the parameters. The differences are so minor that they are almost imperceptible.

Even after setting the strength parameter to a very high value, the output still doesn't seem to reflect the given prompt. It appears as though the prompt is not being considered in the generation process.

### Reproduction

```python
from diffusers import FluxPipeline,FluxImg2ImgPipeline
import torch


# the transformer model is from https://huggingface.co/Kijai/flux-fp8/blob/main/flux1-dev-fp8.safetensors
pipeline = FluxPipeline(...)
pipeline.enable_model_cpu_offload()

prompt = "a cat with wing flying in the sky"
num_inference_steps = 25
guidance_scale = 3
seed = 15827082008518028345
image = pipeline(prompt=prompt,num_inference_steps=num_inference_steps,guidance_scale=guidance_scale,
                 generator=torch.Generator(pipeline.device).manual_seed(seed)).images[0]


i2i_pipeline = FluxImg2ImgPipeline(**pipeline.components)
i2i_pipeline.enable_model_cpu_offload()

i2i_prompt = "in a forest"
i2i_seed = 888
strength = 0.7
i2i_image = i2i_pipeline(image=image,prompt=i2i_prompt,strength=strength,num_inference_steps=num_inference_steps,guidance_scale=guidance_scale,
                         generator=torch.Generator(i2i_pipeline.device).manual_seed(i2i_seed)).images[0]
```
image:
![Colab (44)](https://github.com/user-attachments/assets/b589c168-9f11-4b04-8569-2b16f19bb0c9)
i2i_image:
![Colab (45)](https://github.com/user-attachments/assets/5b57f54f-9c67-43a5-b555-e6de679e897a)

As you can see, the result from image to image pipeline is quite simliar to the input image, however, the strength is 0.7.

### System Info

- 🤗 Diffusers version: 0.32.0.dev0
- Platform: Linux-6.6.56+-x86_64-with-glibc2.35
- Running on Google Colab?: No
- Python version: 3.10.14
- PyTorch version (GPU?): 2.4.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.8.4 (gpu)
- Jax version: 0.4.26
- JaxLib version: 0.4.26.dev20240620
- Huggingface_hub version: 0.26.2
- Transformers version: 4.46.3
- Accelerate version: 1.1.1
- PEFT version: 0.14.0
- Bitsandbytes version: not installed
- Safetensors version: 0.4.5
- xFormers version: not installed
- Accelerator: Tesla T4, 15360 MiB
Tesla T4, 15360 MiB
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@yiyixuxu @DN6

**Comment by SoftologyPro** [comment]
I can confirm this issue, no matter what steps, strength or guidance scale settings I try the resulting image is always very close to the init image.

**Comment by CyberVy** [comment]
@sayakpaul Somebody came across the same issue. Could you please take a look? Thank you!

**Comment by CyberVy** [comment]
> I can confirm this issue, no matter what steps, strength or guidance scale settings I try the resulting image is always very close to the init image.

@SoftologyPro Thanks for your response. It's been a month since I posted this issue, but no one replied. I almost thought no one have the same problem.

**Comment by CyberVy** [comment]
@SoftologyPro Are you still bothered by this bug? It seems it has been fixed in the latest version.

**Comment by SoftologyPro** [comment]
> [@SoftologyPro](https://github.com/SoftologyPro) Are you still bothered by this bug? It seems it has been fixed in the latest version.

Still the same behavior here.  Using the latest diffusers.
Can you share a script that img2img works for you so I can see what may be different with mine?

**Comment by asomoza** [comment]
I remember that when I tested Flux quite a with img2img the range of the denoising strength was very limited, I don't remember the exact numbers now, by the values where in the range of the high 0.7s and low 0.9.

This is not an issue and AFAIK we didn't do something to fix it in the recent versions, this is something from the model and it's probably because it's a distilled version of the model, maybe you can try with the new Flux finetuned model that can use CFG and test this.

**Comment by SoftologyPro** [comment]
> I remember that when I tested Flux quite a with img2img the range of the denoising strength was very limited, I don't remember the exact numbers now, by the values where in the range of the high 0.7s and low 0.9.
> 
> This is not an issue and AFAIK we didn't do something to fix it in the recent versions, this is something from the model and it's probably because it's a distilled version of the model, maybe you can try with the new Flux finetuned model that can use CFG and test this.

Can you give me a link to the new finetuned model and an example script?

**Comment by CyberVy** [comment]
Yes, I think maybe the distilled model is the reason.

**Comment by SoftologyPro** [comment]
Closed?  It is not fixed or resolved yet?
If it is "maybe the distilled model" can someone please give a link to the correct img2img model and a basic script to show to use it?

**Comment by SoftologyPro** [comment]
> [@SoftologyPro](https://github.com/SoftologyPro) My code is similar with yours. But I don't know the model you used. What is the value of your `strength` parameter? Sometimes, a high `strength` (0.8 - 1) is required, especially if your input image is quite different from your prompt.
> 
> My model is [here](https://huggingface.co/eramth/flux-4bit).
> 
> `num_inference_steps` 20
> 
> `guidance_scale` 3.5
> 
> `prompt` a cat in a forest
> 

OK, that model page only gives a quick text to image example

```
from diffusers import FluxPipeline
import torch
pipeline = FluxPipeline("eramth/flux-4bit",torch_dtype=torch.float16).to("cuda")
image = pipeline(prompt="a cute cat",num_inference_steps=25,guidance_scale=3.5).images[0]
```

Can you show some code to do img2img with that model?

**Comment by CyberVy** [comment]
@SoftologyPro 

I've made the reproduction code from which you can get the same results.
```python
from diffusers import FluxImg2ImgPipeline
from diffusers.utils import load_image
import torch

image = load_image("https://github.com/user-attachments/assets/b589c168-9f11-4b04-8569-2b16f19bb0c9")
pipeline = FluxImg2ImgPipeline.from_pretrained("eramth/flux-4bit",cache_dir="./xfusion_models",torch_dtype=torch.float16).to("cuda")
pipeline.vae.enable_tiling()

kw = {"image":image,"prompt":"a cat in a forest","guidance_scale":3.5,"num_inference_steps":20}
seed_0 = 17502695788712410663
image_0 = pipeline(**kw,strength=0.8,generator=torch.Generator(pipeline.device).manual_seed(seed_0)).images[0]
seed_1 = -2127766496927258660
image_1 = pipeline(**kw,strength=0.9,generator=torch.Generator(pipeline.device).manual_seed(seed_1)).images[0]
seed_2 = 3934501538425762656
image_2 = pipeline(**kw,strength=1.0,generator=torch.Generator(pipeline.device).manual_seed(seed_2)).images[0]
```

**Comment by CyberVy** [comment]
> Closed? It is not fixed or resolved yet? If it is "maybe the distilled model" can someone please give a link to the correct img2img model and a basic script to show to use it?

@SoftologyPro 
Don't worry, you can test with my code first, if you still have the issue, we can reopen the issue anytime.

**Comment by SoftologyPro** [comment]
> [@SoftologyPro](https://github.com/SoftologyPro)
> 
> I've made the reproduction code from which you can get the same results.
> 
> First make sure `bitsandbytes` installed, because my model is a quantized model.
> 
> ```
> pip install bitsandbytes
> ```
> 
> from diffusers import FluxImg2ImgPipeline
> from diffusers.utils import load_image
> import torch
> 
> image = load_image("https://github.com/user-attachments/assets/b589c168-9f11-4b04-8569-2b16f19bb0c9")
> pipeline = FluxImg2ImgPipeline.from_pretrained("eramth/flux-4bit",torch_dtype=torch.float16).to("cuda")
> pipeline.vae.enable_tiling()
> 
> kw = {"image":image,"prompt":"a cat in a forest","guidance_scale":3.5,"num_inference_steps":20}
> seed_0 = 17502695788712410663
> image_0 = pipeline(**kw,strength=0.8,generator=torch.Generator(pipeline.device).manual_seed(seed_0)).images[0]
> seed_1 = -2127766496927258660
> image_1 = pipeline(**kw,strength=0.9,generator=torch.Generator(pipeline.device).manual_seed(seed_1)).images[0]
> seed_2 = 3934501538425762656
> image_2 = pipeline(**kw,strength=1.0,generator=torch.Generator(pipeline.device).manual_seed(seed_2)).images[0]

Thank you for the clarifications and code.  It works for me now.
My main goal is to use this for recursive movie making, ie generate a frame, stretch the image slightly, feed it into the next frame as the seed image, repeat for as long as needed.  Using Playgroundv2 for this has worked well.  https://www.instagram.com/softology/ Hopefully I can get some new interesting results with Flux.

**Comment by asomoza** [comment]
[This one](https://huggingface.co/ostris/OpenFLUX.1) is one of the `undistilled` models, I remember there was a couple more but you'll have to look for them, don't really know the quality or how well they work because testing Flux takes a long time and I haven't had the bandwidth to do it. 

Also just so you know, most of the undistilled models are based on schnell so they also have the added feature that they can be used in commercial products.

I'll try to do some tests when I have time.
